connection.go

```diff
@@ -258,6 +258,7 @@ func (c *connection) Close() error {
 // APIVersions sends a request to fetch the supported versions for each API.
 // Versioning is only supported in Kafka versions above 0.10.0.0
 func (c *connection) APIVersions(req *proto.APIVersionsReq) (*proto.APIVersionsResp, error) {
+	req.Version = c.getBestVersion(proto.APIVersionsReqKind)
```

Comment:
Is this sort of mutation/side-effect of passed pointers a usual pattern in this library? What's the background for this here?

Reply:
I think it safe here.   The way a client uses this API is that he creates a Request, passes it to function which returns Response.   By protocol the `Version` field is a part of the Request, but actually it is a settings of a connection.  So when user creates Request he has no idea which version it should be.  When we pass this Request to connection it copies Version setting defined for this connection to Request and send it to server.

Reply:
So, wouldn't it be better to just set the Version once at the connection level, instead of on each request?

Reply:
It's the consequense of design mistake.  The Version field shouldn't be public.  User of our library shouldn't ever set, change or read this field.   The only reason why I cannot make is private is that it defined in package `kafka/proto` and used from connection struct which defined in `kafka`.